src/Streamly/Internal/Network/Socket.hs

```diff
@@ -151,7 +151,7 @@ initListener listenQLen SockSpec{..} addr =
     where
 
     use sock = do
-        mapM_ (\(opt, val) -> setSocketOption sock opt val) sockOpts
+        mapM_ (uncurry (setSocketOption sock)) sockOpts
```

Comment:
I think not using `uncurry` is a bit clearer.

Reply:
Will add `Uncurry` to hlint ignore rules.